search/search_service/proxy/es_proxy_v2_1.py

```diff
@@ -0,0 +1,319 @@
+        'feature_name': 'name.keyword',
+        'entity': 'entity.keyword',
+        'status': 'status.keyword',
+        'tags': 'tags.keyword',
+        'badges': 'badges.keyword',
+    }
+
+    USER_MAPPING = {
+        'full_name': 'name.keyword',
+        'email': 'key',
+        'first_name': 'first_name',
+        'last_name': 'last_name',
+        'resource_type': 'resource_type',
+    }
+
+    RESOUCE_TO_MAPPING = {
+        Resource.TABLE: TABLE_MAPPING,
+        Resource.DASHBOARD: DASHBOARD_MAPPING,
+        Resource.FEATURE: FEATURE_MAPPING,
+        Resource.USER: USER_MAPPING,
+    }
+
+    def __new__(cls: Any,
+                host: str,
+                user: str,
+                password: str,
+                client: Elasticsearch,
+                page_size: int) -> Any:
+        elasticsearch_client = None
+        if client:
+            elasticsearch_client = client
+        else:
+            http_auth = (user, password) if user else None
+            elasticsearch_client = Elasticsearch(host, http_auth=http_auth)
+
+        # check if any index uses the most up to date mappings (version == 2)
+        indices = elasticsearch_client.indices.get_alias(index='*')
+        mappings_up_to_date = False
+        for index in indices:
+            index_mapping = elasticsearch_client.indices.get_mapping(index=index).get(index)
+            mapping_meta_field = index_mapping.get('mappings').get('_meta')
+            if mapping_meta_field is not None and mapping_meta_field.get('version') == 2:
+                mappings_up_to_date = True
+                break
+
+        if mappings_up_to_date:
+            # Use ElasticsearchProxyV2_1 if indexes are up to date with mappings
+            obj = super().__new__(ElasticsearchProxyV2_1)
+            obj.__init__(host=host,
+                         user=user,
+                         password=password,
+                         client=elasticsearch_client,
+                         page_size=page_size)
+            return obj
+
+        # If old mappings are used proxy client should be ElasticsearchProxyV2
+        obj = super().__new__(ElasticsearchProxyV2)
+        obj.__init__(host=host,
+                     user=user,
+                     password=password,
+                     client=elasticsearch_client,
+                     page_size=page_size)
+        return obj
+
+    def __init__(self, *,
+                 host: str = None,
+                 user: str = '',
+                 password: str = '',
+                 client: Elasticsearch = None,
+                 page_size: int = 10) -> None:
+        super().__init__(host=host,
+                         user=user,
+                         password=password,
+                         client=client,
+                         page_size=page_size)
+
+    def get_index_alias_for_resource(self, resource_type: Resource) -> str:
+        resource_str = resource_type.name.lower()
+        alias_config = current_app.config.get(
```

Comment:
can we do this once at init.

Reply:
You mean get a mapping of all aliases for all resources at init?

Reply:
ya